packages/pdfshift/src/Adaptor.js

```diff
@@ -0,0 +1,196 @@
+ * @property {object} errors - Map of errorCodes -> error messages, ie, `{ 404: 'Resource not found;' }`. Pass `false` to suppress errors for this code.
+ * @property {object} form - Pass a JSON object to be serialised into a multipart HTML form (as FormData) in the body.
+ * @property {object} query - An object of query parameters to be encoded into the URL.
+ * @property {object} headers - An object of headers to append to the request.
+ * @property {string} parseAs - Parse the response body as json, text or stream. By default will use the response headers.
+ * @property {number} timeout - Request timeout in ms. Default: 300 seconds.
+ * @property {object} tls - TLS/SSL authentication options. See https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions
+ */
+
+/**
+ * PDF Body Options object. See [PDFShift documentation](https://docs.pdfshift.io/#convert) for more details.
+ * @typedef {Object} PDFBodyOptions
+ * @property {boolean} sandbox - Generates PDF documents in dev mode that doesn't count in the credits.
+ * @property {boolean} encode - Return the generated PDF in Base64 encoded format, instead of raw.
+ */
+
+/**
+ * Generate a PDF from an HTML string.
+ * @param {Function} htmlTemplateString - A HTML string to convert to PDF
+ * @param {PDFBodyOptions} [options] - Optional configuration for PDF generation
+ * @example <caption>Generate a PDF from a HTML string</caption>
+ * generatePDF(
+ * `<html>
+ *   <body>
+ *     <h1>Sales Report</h1>
+ *     <p>Date: 2025-02-01</p>
+ *     <p>Total Sales: $42</p>
+ *   </body>
+ * </html>
+ * `);
+ * @example <caption>Generate a PDF with options</caption>
+ * generatePDF(
+ * `<html>
+ *   <body>
+ *     <h1>Sales Report</h1>
+ *     <p>Date: 2025-02-01</p>
+ *     <p>Total Sales: $42</p>
+ *   </body>
+ * </html>
+ * `, {
+ *   sandbox: true,
+ *   encode: true);
+ * @returns {Operation}
+ */
+export function generatePDF(htmlTemplateString, options) {
+  return async state => {
+    const [resolvedhtmlTemplateString, resolvedOptions] = expandReferences(
+      state,
+      htmlTemplateString,
+      options
+    );
+
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      '/convert/pdf',
+      {
+        body: {
+          source: resolvedhtmlTemplateString,
+          ...resolvedOptions,
+        },
+        parseAs: 'text',
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * PDF Body Options object. See [PDFShift documentation](https://docs.pdfshift.io/#convert) for more details.
+ * @typedef {Object} BodyOptions
+ * @property {string} source - The HTML string to convert to PDF.
+ * @property {boolean} sandbox - Generates PDF documents in dev mode that doesn't count in the credits.
+ * @property {boolean} encode - Return the generated PDF in Base64 encoded format, instead of raw.
+ */
+
+/**
+ * Make a POST request to PDFShift
+ * @example <caption> Convert a HTML string to PDF</caption>
+ * post(
```

Comment:
PDFShifts API only have two endpoints `POST: convert/pdf` and `GET: credits/usage`. Is there a value adding `post()` and `request()` ?

Also we have `util` export from common, i would imagine if someone needs a http function to use `util.post()` or `util.get()`

But again `util` http helpers don't handle authentication for you so maybe it's not a good example 

Perhaps Joe have an opinion on this

Reply:
I wouldn't have added the HTTP helpers for such a simple API, no. Better to focus on the one thing we want, which is generating a pdf

But I don't think I'd remove them. Since Hunter's added (and tested?) them, let's just leave them in place.